packages/peregrine/lib/talons/CheckoutPage/CheckoutError.js

```diff
@@ -0,0 +1,42 @@
+const paymentErrorMessage =
+    'Unable to place order: Transaction has been declined. Please try again later.';
+
+class CheckoutError extends Error {
+    constructor(gqlError, ...params) {
+        super(params);
+        this.name = 'CheckoutError';
+        this.message = gqlError.message;
+        this.error = gqlError;
+    }
+
+    isPaymentInfoError = () => {
+        return this.paymentHasExpired();
+    };
+
+    hasPaymentExpired = () => {
+        /**
+         * TODO
+         *
+         * This is temp stuff, need to have a better check.
+         */
+        return this.error.graphQLErrors.map(({ message }) =>
+            message.includes(paymentErrorMessage)
```

Comment:
Maybe it doesn't matter that the payment is specifically expired. I'm not sure of all the ways that the specific error could be thrown but perhaps we don't even need a custom error class. With GQL mutations we can handle errors as they occur which means we won't ever be using this class for handling something like a `submitShippingMethodError` since that error will be handled by the mutation invoker, `useShippingMethod.js`.

Reply:
There are different errors that can happen during the place order mutation and some of them are not related to payment information. Check this out: https://devdocs.magento.com/guides/v2.3/graphql/mutations/place-order.html#errors

Reply:
Oh nice find on the doc! May as well add a link in a comment at the top of this file.

Also funny enough, the `transaction was declined` error isn't on that page 😆

Reply:
We may have to refine/add new error checks to extend this error class to other checkout page steps.